packages/wallets/src/features/cfd/screens/MT5TradeScreen/MT5TradeLink/MT5TradeLink.tsx

```diff
@@ -15,25 +15,36 @@ type TMT5TradeLinkProps = {
 };
 
 const MT5TradeLink: FC<TMT5TradeLinkProps> = ({ app = 'linux', isDemo = false, platform, webtraderUrl = '' }) => {
-    const { data: ctraderToken } = useCtraderServiceToken();
+    const { refetch } = useCtraderServiceToken();
     const { t } = useTranslation();
     const { icon, link, text, title } = AppToContentMapper[app];
 
-    const onClickWebTerminal = () => {
+    const getToken = async () => {
+        const cTraderTokenResponse = await refetch();
+        if (!cTraderTokenResponse.error) {
+            return cTraderTokenResponse.data?.service_token?.ctrader?.token;
+        }
+    };
+
+    const onClickWebTerminal = async () => {
         const { isStaging, isTestLink } = getPlatformFromUrl();
-        let url;
+        let url, cTraderURL, cTraderToken;
         switch (platform) {
             case CFD_PLATFORMS.DXTRADE:
                 url = isDemo ? 'https://dx-demo.deriv.com' : 'https://dx.deriv.com';
                 break;
             case CFD_PLATFORMS.CTRADER:
-                url = isTestLink || isStaging ? 'https://ct-uat.deriv.com' : 'https://ct.deriv.com';
-                if (ctraderToken) url += `?token=${ctraderToken}`;
+                await getToken().then(token => (cTraderToken = token));
+                cTraderURL = isTestLink || isStaging ? 'https://ct-uat.deriv.com' : 'https://ct.deriv.com';
```

Comment:
URL variables are added in another task and these will be replaced with them after merging that card